### PR TITLE
[jit] Add string index/slice operations

### PR DESCRIPTION
**Pull request by driazati**
Adds support for string indexing (`"a"[0]`) and slicing (`"abc"[1:3]`)
to script.

### CODE REVIEW

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -1298,6 +1298,13 @@ Node* Graph::createDictIndex(Value* dict, Value* index) {
   return n;
 }
 
+Node* Graph::createStringIndex(Value* string, Value* index) {
```
Comment (suo) [inline torch/csrc/jit/ir.cpp]: I don't think this is necessary, you can just the schematized insert, right? We should only add new `createX` methods for unschematized/special nodes.

File: torch/csrc/jit/script/compiler.cpp [hunk 2]
```diff
@@ -2583,6 +2584,11 @@ struct to_ir {
         ->output();
   }
 
+  Value* emitStringIndex(const SourceRange& loc, Value* string, Value* index) {
+    auto string_type = string->type()->cast<StringType>();
```
Comment (suo) [inline torch/csrc/jit/script/compiler.cpp]: if you have a schematized insert, this function is not necessary. Also, I think this line is unused

File: torch/csrc/jit/register_prim_ops.cpp [hunk 3]
```diff
@@ -957,6 +957,31 @@ int64_t normalizeIndex(int64_t idx, int64_t list_size) {
...
+  end = std::min(size, normalizeIndex(end, size));
+
+  if (end <= start) {
+    // Slice is empty
+    push(stack, std::string(""));
+    return 0;
+  }
+
+  std::string result;
+  result.insert(result.begin(), string.begin() + start, string.begin() + end);
```
Comment (suo) [inline torch/csrc/jit/register_prim_ops.cpp]: you don't need an extra `insert()`, just use the constructor e.g. `std::string result(string, start, end-start)`

**Review by suo — APPROVED** [review]
lgtm

File: torch/csrc/jit/script/compiler.cpp [hunk 4]
```diff
@@ -402,6 +402,8 @@ struct Environment {
           {"min", std::make_shared<BuiltinFunction>(prim::min, at::nullopt)},
           {"max", std::make_shared<BuiltinFunction>(prim::max, at::nullopt)},
           {"list", std::make_shared<BuiltinFunction>(aten::list, at::nullopt)},
+          {"ord", std::make_shared<BuiltinFunction>(aten::ord, at::nullopt)},
+          {"rangelist", std::make_shared<BuiltinFunction>(prim::rangelist, at::nullopt)},
```
Comment (fmassa) [inline torch/csrc/jit/script/compiler.cpp]: Nit: is this line a duplicate of the line below?

**Pull request closed without merging**